pkg/sql/bulkmerge/merge_processor.go

```diff
@@ -156,8 +165,117 @@ func (m *bulkMergeProcessor) handleRow(row rowenc.EncDatumRow) (rowenc.EncDatumR
 
 // Start implements execinfra.RowSource.
 func (m *bulkMergeProcessor) Start(ctx context.Context) {
-	m.StartInternal(ctx, "bulkMergeProcessor")
-	m.input.Start(ctx)
+	ctx = m.StartInternal(ctx, "bulkMergeProcessor")
+	// TODO(annie): also find a way to process input/get task from coordinator
+	if err := m.mergeSSTs(ctx, 0); err != nil {
+		// TODO(annie): need to do this?
+		m.MoveToDraining(err)
+	}
+	m.MoveToDraining(nil)
+}
+
+func getKeysForTask(task int, splits []roachpb.Key) (roachpb.Key, roachpb.Key) {
+	if task == 0 {
+		return nil, splits[0]
+	}
+	last := len(splits) - 1
+	if task == last {
+		return splits[task], nil
+	}
+	if task > last {
+		panic(errors.AssertionFailedf("bad"))
+	}
+	return splits[task-1], splits[task]
+}
+
+// This is stolen from roachpb/data.go where we check if spans overlap.
+func isOverlapping(startKey, endKey roachpb.Key, sst execinfrapb.BulkMergeSpec_SST) bool {
+	otherStartKey, otherEndKey := sst.StartKey, sst.EndKey
```

Comment:
nit: you can simplify this because we can assume there is always a startKey and endKey.